### ISSUE TITLE
Usage of mosquitto_callback_register is entirely unclear

### ISSUE THREAD

**Issue by jplatte** [eclipse/mosquitto#2324]
It seems like with a v4 (auth) plugin I can't be notified when a client disconnects to discard client-specific data that I stored in the plugin userdata. The plugin API v5 seems to fix this by allowing one to `mosquitto_callback_register` a callback for disconnect events, but I can't find any documentation on what to pass for `event_data`, nor what `MOSQ_FUNC_generic_callback` is exactly.

**Comment by jplatte** [comment]
Alternatively, can I store client-specific plugin v4 data in the userdata of the `mosquitto` instance passed to the plugin functions? This seems like it couldn't work for setups where multiple (non-collaborating) plugins are loaded, but currently I only care about a single-plugin setup.

**Comment by ralight** [comment]
@jplatte You're right, this is a significant missing part, and I'm aware it needs writing. There are some examples here: https://github.com/eclipse/mosquitto/tree/master/plugins

**Comment by Salathaniel** [comment]
Camera hotel gpd
